osquery/tables/system/darwin/shared_folders.mm

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <OpenDirectory/OpenDirectory.h>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+Status genSharePointEntries(std::vector<NSDictionary *> &sharepoints) {
```

Comment:
I think this should be a `std::vector<std::map<string, string>>` personally. That way, no trace of the Objective-C++ makes it past the implementation of this function.

This would also simplify your genSharedFolders function to:

```cpp
QueryData genSharedFolders(QueryContext &context) {
  auto nameKey = "dsAttrTypeNative:smb_name";
  auto pathKey = "dsAttrTypeNative:directory_path"
  std::vector<std::map<string, string>> sharepoints;
  QueryData results;
  genSharePointEntries(sharepoints);
  for (const auto &sharepoint : sharepoints) {
    Row r;
    r["name"] = sharepoint.find(nameKey) != sharepoint.end() ? sharepoint.at(nameKey) : ""
    r["path"] = sharepoint.find(pathKey) != sharepoint.end() ? sharepoint.at(pathKey) : ""
    results.push_back(r);
  }
  return results;
}
```